### PR TITLE
add RecModel for retrieval

### PR DESCRIPTION
**Pull request by Intsigstephon**
1. support specify stop_layer
2. support specify embedding_size
3. support 4 head types
4. output  include features and logits

### CODE REVIEW

File: ppcls/arch/__init__.py [hunk 1]
```diff
@@ -18,11 +18,14 @@
 import paddle.nn as nn
 
 from . import backbone
+from . import head
 
 from .backbone import *
-from ppcls.arch.loss_metrics.loss import *
+from .head  import *
```
Comment (cuicheng01) [inline ppcls/arch/__init__.py:24]: 需要用的模块import进来吧，尽量不要使用*了吧

File: ppcls/configs/ImageNet/ResNet/ResNet50.yaml [hunk 2]
```diff
@@ -16,7 +16,21 @@ Global:
...
+  # Backbone:
+  #   name: "ResNet50"
+  # Stoplayer: 
+  #   name: "flatten_0"
+  #   output_dim: 2048
+  #   embedding_size: 512
+  # Head:
+  #   name: "ArcMargin"
+  #   margin: 0.5
+  #   scale:  80
```
Comment (weisy11) [inline ppcls/configs/ImageNet/ResNet/ResNet50.yaml]: 不要直接改这个配置文件，添加一个新的示例吧。

**Review by weisy11 — APPROVED** [review]
LGTM
